### ISSUE TITLE
[BUG] False positives in rules with `FIELD_DECLARATION` location

### ISSUE THREAD

**Issue by abrugaro** [konveyor/analyzer-lsp#1008]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

main

### Priority

Undefined (Default)

### Current Behavior

The `JMS' Topic must be replaced with an Emitter` [rule](https://github.com/konveyor/rulesets/blob/b34502f33c13b9e6d24240e30288dfda7b41368e/default/generated/quarkus/218-jms-to-reactive-quarkus.windup.yaml#L131) with id `jms-to-reactive-quarkus-00040` is defined as follows:

```yaml
ruleID: jms-to-reactive-quarkus-00040
  when:
    or:
    - java.referenced:
        location: FIELD_DECLARATION
        pattern: javax.jms.Topic
    - java.referenced:
        location: FIELD_DECLARATION
        pattern: jakarta.jms.Topic
```
Reports issues in:

- https://github.com/konveyor-ecosystem/coolstore/blob/main/src/main/java/com/redhat/coolstore/service/ShoppingCartOrderProcessor.java#L24 (OK)

- https://github.com/konveyor-ecosystem/coolstore/blob/main/src/main/java/com/redhat/coolstore/service/ShoppingCartOrderProcessor.java#L8

The last issue reported is not a field declaration but an import. There are rules that specifically detect references to the javax.jms.Topic package ([jms-to-reactive-quarkus-00050](https://github.com/konveyor/rulesets/blob/b34502f33c13b9e6d24240e30288dfda7b41368e/default/generated/quarkus/218-jms-to-reactive-quarkus.windup.yaml#L152))

This issue is reproducible in KAI and the operator while analyzing the coolstore app to quarkus
